app/views/concerns/facet_presenter.rb

```diff
@@ -0,0 +1,127 @@
+module FacetPresenter
+  extend ActiveSupport::Concern
+
+  protected
+
+  def facet_item_url(facet, item)
+    if facet_in_params?(facet, item)
+      search_action_url(remove_facet_params(facet, item, params))
+    else
+      search_action_url(add_facet_params_and_redirect(facet, item))
+    end
+  end
+
+  def facet_display(facet, show_children = false)
```

Comment:
Cyclomatic complexity for facet_display is too high. [7/6]<br>Assignment Branch Condition size for facet_display is too high. [16.19/15]<br>Method has too many lines. [12/10]<br>Perceived complexity for facet_display is too high. [9/7]